### PR TITLE
Update to arrow 38

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/src/physical_plan/sorts/sort.rs [hunk 1]
```diff
@@ -1089,7 +1089,7 @@ mod tests {
     #[tokio::test]
     async fn test_sort_fetch_memory_calculation() -> Result<()> {
         // This test mirrors down the size from the example above.
-        let avg_batch_size = 6000;
+        let avg_batch_size = 5000;
```
Comment (tustvold) [inline datafusion/core/src/physical_plan/sorts/sort.rs:1092]: Arrays are smaller because of https://github.com/apache/arrow-rs/pull/4061

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @tustvold

**Comment by alamb** [comment]
🤔  there appear to be more changes needed to tests in other modules

File: datafusion/core/tests/sqllogictests/test_files/arrow_typeof.slt [hunk 2]
```diff
@@ -303,7 +303,7 @@ select arrow_cast('30 minutes', 'Interval(MonthDayNano)');
 
 ## Duration
 
-query error DataFusion error: Error during planning: Cannot automatically convert Interval\(MonthDayNano\) to Duration\(Second\)
+query error DataFusion error: This feature is not implemented: Can't create a scalar from array of type "Duration\(Second\)"
```
Comment (tustvold) [inline datafusion/core/tests/sqllogictests/test_files/arrow_typeof.slt:306]: I think this is the result of https://github.com/apache/arrow-rs/pull/4020

**Review by alamb — APPROVED** [review]
LGTM -- thank you @tustvold

**Pull request merged**